SignalServiceKit/src/Util/NSUserDefaults+OWS.m

```diff
@@ -0,0 +1,38 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "NSUserDefaults+OWS.h"
+#import "TSConstants.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation NSUserDefaults (OWS)
+
++ (NSUserDefaults *)appUserDefaults
+{
+    return [[NSUserDefaults alloc] initWithSuiteName:SignalApplicationGroup];
+}
+
++ (void)migrateToSharedUserDefaults
+{
+    NSUserDefaults *appUserDefaults = self.appUserDefaults;
+
+    NSDictionary<NSString *, id> *dictionary = [NSUserDefaults standardUserDefaults].dictionaryRepresentation;
+    for (NSString *key in dictionary) {
+        id value = dictionary[key];
+        OWSAssert(value);
+        [appUserDefaults setObject:value forKey:key];
+    }
+}
+
++ (void)removeAll
+{
+    NSString *appDomain = NSBundle.mainBundle.bundleIdentifier;
+    [NSUserDefaults.standardUserDefaults removePersistentDomainForName:appDomain];
```

Comment:
Do we need to:
`[[NSUserDefaults standardUserDefaults] synchronize];`

Reply:
Good idea - and I've reworked how we clear out these defaults too.